cmd/nri-kubernetes/main.go

```diff
@@ -234,15 +232,23 @@ func buildClients(c *config.Config) (*clusterClients, error) {
 
 	var ksmCli *ksmClient.Client
 	if c.KSM.Enabled {
-		ksmCli, err = ksmClient.New(ksmClient.WithLogger(logger))
+		ksmCli, err = ksmClient.New(
+			ksmClient.WithLogger(logger),
+			ksmClient.WithTimeout(c.KSM.Timeout),
+			ksmClient.WithMaxRetries(c.KSM.Retries),
+		)
 		if err != nil {
 			return nil, fmt.Errorf("building KSM client: %w", err)
 		}
 	}
 
 	var kubeletCli *kubeletClient.Client
 	if c.Kubelet.Enabled {
-		kubeletCli, err = kubeletClient.New(kubeletClient.DefaultConnector(k8s, c, k8sConfig, logger), kubeletClient.WithLogger(logger))
+		kubeletCli, err = kubeletClient.New(
+			kubeletClient.DefaultConnector(k8s, c, k8sConfig, logger),
+			kubeletClient.WithLogger(logger),
+			kubeletClient.WithMaxRetries(c.Kubelet.Retries),
```

Comment:
why not kubeletClient.WithTimeout?

Reply:
Because the timeout is on the connector, the client uses the timeout from the httpClient in the connector

Reply:
The retries instead, are owned by Pester.

Reply:
As I could find out, Pester's timeout won't take place if an HttpClient is passed to it, as it's the case in the Kubelet's and ControlPlane's Clients where it's got from the connector. So the timeout needs to be set there. I thought about moving the pester creation to the connectors, but found out that too many things needed to be modified and would be a huge refactor.

Reply:
oh true, I completely forgot even if we talked about it yesterday. 😄